R/html_dependency.R

```diff
@@ -231,32 +227,39 @@ copyDependencyToDir <- function(dependency, outputDir, mustWork = TRUE) {
     }
   }
 
-  if (!file.exists(outputDir))
+  if (!dir_exists(outputDir))
     dir.create(outputDir)
 
   target_dir <- file.path(outputDir,
     paste(dependency$name, dependency$version, sep = "-"))
 
-  if (!file.exists(target_dir)) {
-    dir.create(target_dir)
+  # completely remove the target dir because we don't want possible leftover
+  # files in the target dir, e.g. we may have lib/foo.js last time, and it was
+  # removed from the original library, then the next time we copy the library
+  # over to the target dir, we want to remove this lib/foo.js as well
+  if (dir_exists(target_dir)) unlink(target_dir, recursive = TRUE)
```

Comment:
`unlink(recursive = TRUE)` makes my stomach hurt... just for peace of mind, can we do some sanity checks here, e.g. make sure that dependency$name and dependency$version are length 1 and not `""`, and make sure `outputDir` is non-`""`, non-`"/"`, etc.?

I'm sure we can prove that this particular snapshot of the code is correct, but more checks and comments with exclamation points will help protect us as the code evolves over time.

https://www.google.com/webhp?hl=en#hl=en&q=rm+rf+bug